CONTRIBUTING.md

```diff
@@ -65,10 +65,58 @@ the [puppet-dev mailing list](https://groups.google.com/forum/#!forum/puppet-dev
       why this is a problem, and how the patch fixes the problem when applied.
   ```
 * Make sure you have added the necessary tests for your changes.
-* Run _all_ the tests to assure nothing else was accidentally broken. First
-  install all the test dependencies with `bundle install --path .bundle`. Then
-  either run all the tests serially with `bundle exec rspec spec` or in parallel
-  with `bundle exec rake parallel:spec[process_count]`
+
+## Running Tests
+Before you submit your pull request, please ensure all unit tests pass. This
+includes any tests you may have added or modified in addition to all existing
+unmodified tests.
+
+The general workflow is as follows:
+* Ensure ruby and bundler are installed
+* Install dependencies specified with bundler
+  `bundle install --path .bundle`
+* Run the tests. You have a few different options on how to run them:
+  * Run one test by specifying the path to the test
+    `bundle exec rspec spec/unit/puppet_spec.rb`
+  * Run all tests serially
+    `bundle exec rspec`
+  * Run all tests in parallel. Here `4` is the number of processors to use when
+  running tests. This number will be different based on where you are running
+  these tests.
+    `bundle exec rake parallel:spec[4]`
+
+#### On Windows
+To run the spec tests on windows, you'll want to use PowerShell to start. Please
+note you likely will not be able to simply copy and paste the below script. It
+is meant as a guide to help you get tests running on windows. Some of the
+commands below are interactive and will require input or additional clicks from
+you.
+```
+# Install chocolatey
+Set-ExecutionPolicy Bypass -Scope Process -Force; iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))
+# Install ruby 2.4. As of this writing (09-2018), Ruby 2.4 is the latest version
+# of ruby that we support. The latest available through chocolatey is too new
+# and will break tests.
+choco install -y ruby --version 2.4.3.1
+# Add ruby to the path. If you installed a version other than Ruby 2.4, be sure
+# to verify if you need to use a slightly different path than the one below
+$env:Path += ";C:\tools\ruby24\bin"
+# Install MSYS2 and MINGW development toolchain. When asked which components to
+# install, select the third option. You need both MSYS2 and MINGW to compile the
+# native dependencies installed with bundler.
+ridk install
+# Install bundler
+gem install bundler
+# Clone puppet and cd into the repo
+git clone https://github.com/puppetlabs/puppet.git
+cd puppet
+# Install dependencies
+bundle install
+# Run all of the tests, where 4 is the number of processors to use
+bundle exec rake parallel:spec[4]
+# Run a specific test
```

Comment:
If you wanted to go one step further, you can specify line number for just one test, otherwise I'd say this is more likely to be a group of tests, or some other word artistry